test/test_type_promotion.py

```diff
@@ -205,6 +205,9 @@ def test_many_promotions(self, device):
                 if op == torch.sub and common_dtype != torch.bool:
                     # Subtraction, the `-` operator, with a bool tensor is not supported.
                     continue
+                # TODO: skip this short-circuit once integer division is reenabled for div
+                if op == torch.div and common_dtype not in (torch.half, torch.float, torch.double):
```

Comment:
this looks fine if you want to do it this way, but it might be less churn to just run `true_divide` through this instead of `div` and make the TODO change `true_divide` back into `div`.  Then you don't actually need to change any of the logic or remember what changes you have to undo.

Reply:
I also think these test changes should be separate.  Especially because this is a BC breaking PR, we really want it to be minimally standalone in case it needs to be reverted (or even figuring out for sure what the BC breaking change was).

Reply:
Moved out of this PR. Good idea to minimize change.